classes/helper.php

```diff
@@ -549,6 +566,57 @@ public static function search_db(\tool_advancedreplace\search $record, string $o
         }
     }
 
+    /**
+     * Return a closure that can be used to create the link from the record.
+     *
+     * @param string $table   The name of the table being searched.
+     * @param string $column  The name of the column being searched.
+     * @return \Closure    $urlstring = closure($record).
+     */
+    static private function find_link_function($table, $column) {
+
+        $linktypes = [
+            'course' => function($record) {
+                return (new \moodle_url('/course/view.php', array('id' => $record->id)))->out();
+            },
+            ];
+
+        static $linkmappings = [
+            'course:fullname' => 'course',
+            'course:shortname' => 'course',
+            'course:summary' => 'course',
+        ];
+
+        if (! empty($linkmappings["{$table}:{$column}"])) {
+            $type = $linkmappings["{$table}:{$column}"];
+            if (! empty($linktypes[$type]) ) {
+                $linkfunction = $linktypes[$type];
+                return $linkfunction;
+            }
+        }
+
+        // Next consider links based on the table name being a module.
+        static $modulefunctions = null;
```

Comment:
I find the order here a bit confusing to follow, generally all the static vars at at the top and if they are set then you can exit early and avoid lots of code inside nested ifs